### ISSUE TITLE
Suggested fixes for gemini

### ISSUE THREAD

**Issue by jl3676** [jdf-prog/LLM-Engines#1]
While using LLM-Engines, which is a great library that saved me a lot of time, I encountered a few issues with the gemini API, which I worked around by modifying the source code. I'll list the fixes I made below for reference.

Problem 1: 

Currently, Gemini's API doesn't take the system prompt through the system role in the input messages. The following lines in `gemini.py` caused an error when I ran them:
```
model = genai.GenerativeModel(model_name)

new_messages = []
if conv_system_msg:
    new_messages.append({"role": "system", "parts": [glm.Part(text=conv_system_msg)]})
for i, message in enumerate(messages):
    new_messages.append({"role": "user" if i % 2 == 0 else "model", "parts": [glm.Part(text=message)]})
```
I changed these lines to:
```
model = genai.GenerativeModel(model_name, system_instruction=conv_system_msg)

new_messages = []
for i, message in enumerate(messages):
    new_messages.append({"role": "user" if i % 2 == 0 else "model", "parts": [glm.Part(text=message)]})
```
which worked.

Problem 2:

When the Gemini API blocked my requests, the code was stuck because it assumed that a response text would be returned. However, when the request is blocked, which happened to about 1% of my prompts, the `response` variable does not have a field named `text`. To fix this, I changed this line in `gemini.py`
```
return response.text
```
to
```
try:
    return response.text
except:
    return None
```
which allowed my code to proceed past the blocked queries.

**Comment by jdf-prog** [comment]
Thanks for raising the issue! Can you open a PR so I can merge these changes and you can be one of the contributor?

**Comment by jdf-prog** [comment]
Closed as [PR #2](https://github.com/jdf-prog/LLM-Engines/pull/2) merged.
